pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/BinaryOperator.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.ast;
+
+import apex.jorje.data.ast.AssignmentOp;
+import apex.jorje.data.ast.BinaryOp;
+import apex.jorje.data.ast.BooleanOp;
+
+/**
+ * Apex binary operator
+ */
+public enum BinaryOperator {
```

Comment:
Same thing for these operators. I find it confusing that ASTAssignmentOperator::getOp should return a BinaryOperator when really only a subset of those constants is valid there (the assignment ops). If forces a user to handle the other constants even when they're known not to occur...